src/rcamera.h

```diff
@@ -134,434 +148,401 @@ void SetCameraMoveControls(int keyFront, int keyBack,
+    if (IsKeyDown(KEY_D)) CameraMoveRight(camera, CAMERA_MOVE_SPEED);
+    if (IsKeyDown(KEY_A)) CameraMoveRight(camera, -CAMERA_MOVE_SPEED);
+    if (IsKeyDown(KEY_SPACE)) CameraMoveUp(camera, CAMERA_MOVE_SPEED);
+    if (IsKeyDown(KEY_LEFT_CONTROL)) CameraMoveUp(camera, -CAMERA_MOVE_SPEED);
+
+    // Camera rotation
+    if (IsKeyDown(KEY_DOWN)) CameraPitch(camera, -CAMERA_ROTATION_SPEED);
+    if (IsKeyDown(KEY_UP)) CameraPitch(camera, CAMERA_ROTATION_SPEED);
+    if (IsKeyDown(KEY_RIGHT)) CameraYaw(camera, -CAMERA_ROTATION_SPEED);
+    if (IsKeyDown(KEY_LEFT)) CameraYaw(camera, CAMERA_ROTATION_SPEED);
+    if (IsKeyDown(KEY_Q)) CameraRoll(camera, -CAMERA_ROTATION_SPEED);
+    if (IsKeyDown(KEY_E)) CameraRoll(camera, CAMERA_ROTATION_SPEED);
+
+    CameraYaw(camera, mousePositionDelta.x * -CAMERA_MOUSE_MOVE_SENSITIVITY);
+    CameraPitch(camera, mousePositionDelta.y * -CAMERA_MOUSE_MOVE_SENSITIVITY);
+
+    // Zoom target distance
+    CameraZoom(camera, -GetMouseWheelMove());
+    if (IsKeyPressed(KEY_KP_SUBTRACT)) CameraZoom(camera, 2.0f);
+    if (IsKeyPressed(KEY_KP_ADD)) CameraZoom(camera, -2.0f);
+
+    // Switch camera mode
+    if (IsKeyPressed(KEY_ONE)) SetCameraMode(camera, CAMERA_FREE);
+    if (IsKeyPressed(KEY_TWO)) SetCameraMode(camera, CAMERA_FIRST_PERSON);
+    if (IsKeyPressed(KEY_THREE)) SetCameraMode(camera, CAMERA_THIRD_PERSON);
+    if (IsKeyPressed(KEY_FOUR)) SetCameraMode(camera, CAMERA_ORBITAL);
+
+    // Switch camera projection
+    if (IsKeyPressed(KEY_P)) {
+        if (camera->projection == CAMERA_PERSPECTIVE) {
+            // Create isometric view
+            camera->mode = CAMERA_THIRD_PERSON;
+            // Note: The target distance is related to the render distance in the orthographic projection
+            camera->position = (Vector3){ 0.0f, 2.0f, -100.0f };
+            camera->target = (Vector3){ 0.0f, 2.0f, 0.0f };
+            camera->up = (Vector3){ 0.0f, 1.0f, 0.0f };
+            camera->projection = CAMERA_ORTHOGRAPHIC;
+            camera->fovy = 20.0f; // near plane width in CAMERA_ORTHOGRAPHIC
+            CameraYaw(camera, -135 * DEG2RAD);
+            CameraPitch(camera, -45 * DEG2RAD);
+        }
+        else if (camera->projection == CAMERA_ORTHOGRAPHIC)
+        {
+            // Reset to default view
+             camera->mode = CAMERA_THIRD_PERSON;
+            // Note: The target distance is related to the render distance in the orthographic projection
+            camera->position = (Vector3){ 0.0f, 2.0f, 10.0f };
+            camera->target = (Vector3){ 0.0f, 2.0f, 0.0f };
+            camera->up = (Vector3){ 0.0f, 1.0f, 0.0f };
+            camera->projection = CAMERA_PERSPECTIVE;
+            camera->fovy = 60.0f;
+        }
+    }
+
+    // Apply view bobbing when moving around (per default only active in CAMERA_FIRST_PERSON)
+    if (IsKeyDown(KEY_W) || IsKeyDown(KEY_A) || IsKeyDown(KEY_S) || IsKeyDown(KEY_D)) CameraViewBobbing(camera);
```

Comment:
Tilting and swinging is mode dependant (first person camera) and setting the tilting and swinging to 0 should be enough, it shouldn't depend on the inputs, I mean, it's dependant on the movement.

Reply:
The `CameraViewBobbing(..)` function does nothing if the `swingCounter` is set 0.
It is only activated (set to 1) in the first person mode.

If we remove the mode variable anyway though, and just pass it as a parameter to `UpdateCamera(..)`, then this problem is solved too.